### PR TITLE
When starting up try creating directories recursively

### PR DESCRIPTION
**Pull request by hpvb**
Previously we had a check to see if cache and data directories exist and
another check to try to make them if they do not. However the second
check was never reached if we don't have the directories in question.

Furthermore for cache directories on Linux people who never started a
desktop environment we need to recurisively create the XDG directory as
well as the godot specific directory.

This fixes #17963

### CODE REVIEW

File: editor/editor_settings.cpp [hunk 1]
```diff
@@ -764,21 +764,15 @@ void EditorSettings::create() {
 		}
 
 		if (dir->change_dir("templates") != OK) {
-			dir->make_dir("templates");
+			dir->make_dir_recursive("templates");
```
Comment (akien-mga) [inline editor/editor_settings.cpp]: Unless I'm missing something, `make_dir_recursive()` doesn't seem needed to me here, we've just created the `data_dir` above, and anyway `"templates"` doesn't include the relevant information to recursively make the absolute path to the target folder.

Comment (akien-mga) [reply editor/editor_settings.cpp]: That said it doesn't *hurt* either, I suppose it does exactly the same logic when there's no recursiveness needed.

Comment (hpvb) [reply editor/editor_settings.cpp]: you're right, I'll change it

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Cherry-picked for 3.0.7.

**Pull request merged**